api/source/graphql/queries/CitationQueries.ts

```diff
@@ -10,4 +19,30 @@ export default {
       return db.select().table(CITATION_TABLE_NAME);
     },
   },
-};
+
+  activeCitation: {
+    type: GraphQLList(citationDocuments),
+    resolve: async () => {
+      const today = new Date();
+      const date = today.getFullYear().toString() + '-' + (today.getMonth() + 1).toString().padStart(2, '0') + '-' + today.getDate().toString().padStart(2, '0');
+      
+      const citations = await db.select().table(CITATION_TABLE_NAME).where('end_date', '>=', date);
+
+      const newActiveCitations = await Promise.all(
+        citations.map(async (myCitation) => {
+          const documents = await db.select().table(DOCUMENT_TABLE_NAME).where({ id: myCitation.document_id });
+
+          const newCitation = {
+            citation: myCitation,
+            document: documents[0],
+
+          };
+          return newCitation;
+        }),
+      );
+
+      console.log(newActiveCitations);
+      return [...newActiveCitations]
+    },
+  },
+};
```

Comment:
Weird query. `citationDocuments` should never be created. Adding filters is a better option.

Follow this example to fetch documents.
```
query {
    citations (active: true) {
        documents {
            id
            ...
        }
    }
}
```

Review GraphQL main concepts [HERE](https://www.youtube.com/watch?v=qux4-yWeZvo).